src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4682,6 +4766,19 @@ static TypeWithState calculateResultType(TypeWithAnnotations targetTypeWithNulla
             return operandType;
         }
 
+        private void SnapshotWalkerThroughConversionGroup(BoundConversion conversionOpt, BoundExpression convertedNode)
```

Comment:
>SnapshotWalkerThroughConversionGroup [](start = 21, length = 36)

Is this gymnastics still needed after the switch to position based snapshots?

Reply:
Yes, because there are a couple of cases where we remove both explicit and implicit conversions. I now bail out as soon as we get to a conversion with the same syntax as the underlying node to avoid taking extra snapshots.

---
In reply to: [289493249](https://github.com/dotnet/roslyn/pull/35850#discussion_r289493249) [](ancestors = 289493249)